community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -734,31 +734,15 @@ private void checkpoint( IOLimiter ioLimiter, Header.Writer headerWriter ) throw
     }
 
     /**
-     * Closes this tree and its associated resources. A {@link #checkpoint(IOLimiter)} is first performed
-     * as part of this call if there have been changes since last call to {@link #checkpoint(IOLimiter)}
-     * or since opening this tree.
+     * Closes this tree and its associated resources.
+     * NOTE: No {@link #checkpoint(IOLimiter) checkpoint} is performed. To make data persistent in store
+     * a checkpoint needs to be performed manually.
```

Comment:
That depends on where this tree is used. As a general statement it has to be done manually. In the context of being part of a neo4j database it's done by the integration points inside neo4j that manages these trees.

Comparing with documentation about recovery it's documented how interaction with the rest of the world is supposed to look like, not that "the system takes care of recovery". I'd like the javadocs in here to focus on how others should interact with it.